framework/src/base/FEProblem.C

```diff
@@ -1468,10 +1468,15 @@ void
 FEProblem::addAuxKernel(const std::string & kernel_name, const std::string & name, InputParameters parameters)
 {
   parameters.set<FEProblem *>("_fe_problem") = this;
+
+  if (_displaced_problem != NULL)
```

Comment:
This isn't the right thing to do.  The `_sys` should reflect which system this object is actually working on (as specified by `use_displaced_mesh` and the presence of the `_displaced_problem`)